docs/beamrunner/beamrunner-3-sample.md

```diff
@@ -0,0 +1,172 @@
+
+```
+PCollection<KV<String,Double>> mergedReadings = deviceReadings
+        .apply("MergeReadings", Flatten.<KV<String,Double>>pCollections())
+        .apply("WindowReadings",
+                Window.<KV<String,Double>>into(
+                        FixedWindows.of(Duration.standardSeconds(options.getWindowSize()))));
+```
+
+The windowed data is validated with a ValidateReadings transform that produces two output streams, one of good readings and one of bad readings:
+
+```
+final TupleTag<KV<String,Double>> goodTag = new TupleTag<KV<String,Double>>(){ ... }
+final TupleTag<KV<String,Double>> badTag = new TupleTag<KV<String,Double>>(){ ... }
+final double badTempThreshold = options.getBadTempThreshold();
+
+PCollectionTuple validatedReadings = mergedReadings.apply("ValidateReadings",
+        ParDo.of(new DoFn<KV<String,Double>,KV<String,Double>>() {
+            private static final long serialVersionUID = 1L;
+
+            @ProcessElement
+            public void processElement(ProcessContext c) {
+                String device = c.element().getKey();
+                Double temp = c.element().getValue();
+                if (temp < badTempThreshold) {
+                    // Good reading, output and update distribution metrics
+                    c.output(c.element());
+                    ...
+                } else {
+                    // Bad reading, output to side output, update counters
+                    c.output(badTag, c.element());
+                    ...
+                }
+            }
+        }).withOutputTags(goodTag, TupleTagList.of(badTag)));
+
+
+```
+
+On the good readings, statistics are calculated in a GoodStats transform and then logged by a GoodLog transform:
+
+```
+validatedReadings
+        .get(goodTag)
+        .apply("GoodStats", Combine.<String, Double, Stats>perKey(new DeviceStatsFn()))
+        .apply("GoodLog", ParDo.of(new LogWindowFn<Stats>("Temperature device statistics")));
+```
+
+The bad readings are counted by a BadCount transform and logged by the BadLog transform:
+
+```
+validatedReadings
+        .get(badTag)
+        .apply("BadCount", Count.<String,Double>perKey())
+        .apply("BadLog", ParDo.of(new LogWindowFn<Long>("Temperature device error count")));
+```
+
+The structure of the application becomes clear after it is displayed in the Streams console after the application is run.
+
+## Running the `TemperatureSample` application
+
+1. Make sure that your Streaming Analytics service on Bluemix is running.
+2. Navigate to the `$STREAMS\_RUNNER\_HOME/samples` directory. The Streams Runner toolkit provides all necessary files. Assuming that all environment variables are set as described in Installing the toolkit and that $VCAP\_SERVICES has credentials in it named beam-service, you can launch the `TemperatureSample` application with the following command:
+
+   ```
+   java -cp \

```

Comment:
@pamlane  @pgerv12  This command did not work for me. I had to add
` $STREAMS_INSTALL/lib/com.ibm.streams.operator.jar` to the `-cp` argument. 

I set all the environment variables:
```
[streamsadmin@streamsqse samples]$ env| grep STREAMS
STREAMS_SPLPATH=/opt/ibm/InfoSphere_Streams/4.2.1.1/toolkits
STREAMS_RUNNER_HOME=/home/streamsadmin/Downloads/beam-ga/com.ibm.streams.beam-1.0.0
STREAMS_QSEVM=/opt/ibm/InfoSphere_Streams/4.2.1.1
STREAMS_INSTALL=/opt/ibm/InfoSphere_Streams/4.2.1.1
STREAMS_DOMAIN_ID=StreamsDomain
STREAMS_BEAM_TOOLKIT=/home/streamsadmin/Downloads/beam-ga/com.ibm.streams.beam-1.0.0/com.ibm.streams.beam
STREAMS_INSTANCE_ID=StreamsInstance
[streamsadmin@streamsqse samples]$ env | grep -i streaming
STREAMING_ANALYTICS_SERVICE_NAME=beam-service

```

Reply:
We do specify in the installation page that the `STREAMS_INSTALL` env should be unset if users want to build their application on Bluemix (which is what we want). However, I think it would be good to reiterate that

Reply:
There needs to be some clarification. If you are not supposed to set STREAMS_INSTALL, why does the Installation section list setting it as one of the environment variables?
Following the guide as it is on the QSE leads to an error.

Some clarification is needed in the introduction and in the installation sections to make clear that you do not need Streams installed in order to use the runner. In addition, if you have Streams installed (e.g. the QSE) you need to unset Streams install in order to compile on bluemix, especially if your version of Streams is old.

If we are focusing on those who do not have Streams installed and not documenting how to use the runner outside of Bluemix then we should either remove the comment about a local domain/instance from the introduction.

Reply:
I don't think you should need `operator.jar` for `TemperatureSample` (not sure about others). If you do, that's a problem for pure remote build since we aren't supposed to require `$STREAMS_INSTALL`.

You're not only supposed to *not* set `$STREAMS_INSTALL`, but in some instances, like QSE, you actually have to *unset* it (or set it empty), but either way, we shouldn't be referencing it in the classpath.

Reply:
To clarify  - if you have STREAMS_INSTALL set, then the classpath, as set in the doc, is incomplete and I get a ClassNotFoundException. I resolved that by updating it to add the operator jar. But, once I unset STREAMS_INSTALL, I was able to run the sample using the provided command.

Reply:
@natashadsilva That sounds right, I think the only question is whether we support the first scenario (doing local builds if you have a Streams install) because that adds a bunch of other dependencies we don't articulate in the doc (Streams version, OS version, etc)